File: app/src/main/java/han/ayeon/searchimgandvideo/view/searchlist/list/SearchListViewHolder.java
package han.ayeon.searchimgandvideo.view.searchlist.list;

import android.widget.CheckBox;
import android.widget.ImageView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;

import han.ayeon.searchimgandvideo.R;
import han.ayeon.searchimgandvideo.databinding.FragmentSearchResultItemBinding;
import han.ayeon.searchimgandvideo.model.data.ResultData;
import han.ayeon.searchimgandvideo.view.ImageLoader;
import han.ayeon.searchimgandvideo.view.SavedListChangeCallBackListener;


public class SearchListViewHolder extends RecyclerView.ViewHolder implements ImageLoader {

    private ImageView thumbNailView = itemView.findViewById(R.id.thumbnail_view);
    private CheckBox saveCheckBox = itemView.findViewById(R.id.save_button);


    SearchListViewHolder(@NonNull FragmentSearchResultItemBinding itemView) {

        super(itemView.getRoot());
    }

    @Override
    public void showThumbNail(String url) {
        Glide.with(itemView)
                .load(url)
                .placeholder(R.drawable.ic_search)
                .into(thumbNailView);
    }

    public void setSavedState(boolean isSaved) {
        saveCheckBox.setChecked(isSaved);
    }

    void itemOnClickListener (final SavedListChangeCallBackListener savedListChangeCallBackListener,
                              final ResultData item) {
        saveCheckBox.setOnClickListener(v -> {
            if(item.isSaved()) {
                savedListChangeCallBackListener.removed(item);
                item.setSaved(false);
                saveCheckBox.setChecked(false);
            } else {
                savedListChangeCallBackListener.saved(item);
                item.setSaved(true);
                saveCheckBox.setChecked(true);
            }
        });
    }
}

File: app/src/main/java/han/ayeon/searchimgandvideo/presentation/view/searchlist/SearchListViewHolder.kt
package han.ayeon.searchimgandvideo.presentation.view.searchlist

import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import han.ayeon.searchimgandvideo.R
import han.ayeon.searchimgandvideo.databinding.FragmentSearchResultItemBinding
import han.ayeon.searchimgandvideo.domain.data.Media
import han.ayeon.searchimgandvideo.presentation.view.ImageLoader
import han.ayeon.searchimgandvideo.presentation.viewmodel.SearchResultViewModel
import kotlinx.android.synthetic.main.fragment_search_result_item.view.*

/**
 * Created by HanAYeon on 2019-07-18.
 */

class SearchListViewHolder(itemView: FragmentSearchResultItemBinding) : RecyclerView.ViewHolder(itemView.root), ImageLoader {
    override fun showThumbNail(url: String) {
        Glide.with(itemView)
                .load(url)
                .placeholder(R.drawable.ic_search)
                .into(itemView.thumbnail_view)
    }

    fun setSavedState(isSaved: Boolean) {
        itemView.save_button.isChecked = isSaved
    }

    fun itemOnClickListener(viewModel: SearchResultViewModel, item: Media) {
        itemView.save_button.setOnClickListener {
            if (item.isSaved) {
                viewModel.removedSavedItem(item)
                item.isSaved = false
                itemView.save_button.isChecked = false
            } else {
                viewModel.addSavedItem(item)
                item.isSaved = true
                itemView.save_button.isChecked = true
            }
        }
    }

}